.github/workflows/generate-paradox-check.yml

```diff
@@ -0,0 +1,48 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Generate paradox check
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  generate-doc-check:
+    name: Generate paradox check
+    runs-on: ubuntu-20.04
+    if: github.repository == 'apache/incubator-pekko'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Setup Java 11
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 11
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.4.4
+
+      - name: Compile paradox documents
+        run: sbt ";clean ;Test / compile ;TestJdk9 / compile ;docs/paradox"
```

Comment:
can we do this as part of https://github.com/apache/incubator-pekko/blob/main/.github/workflows/binary-compatibility-checks.yml - to avoid have an extra job?

Reply:
I can move this to `Generate doc check / Generate doc check (pull_request)` task